lib/streamlit/color_util.py

```diff
@@ -0,0 +1,274 @@
+    to _float_formatter and alpha_formatter to _int_formatter.
+    """
+    if is_hex_color_like(color):
+        hex_len = len(color)
+        color_hex = cast(str, color)
+
+        if hex_len == 4:
+            r = 2 * color_hex[1]
+            g = 2 * color_hex[2]
+            b = 2 * color_hex[3]
+            a = "ff"
+        elif hex_len == 5:
+            r = 2 * color_hex[1]
+            g = 2 * color_hex[2]
+            b = 2 * color_hex[3]
+            a = 2 * color_hex[4]
+        elif hex_len == 7:
+            r = color_hex[1:3]
+            g = color_hex[3:5]
+            b = color_hex[5:7]
+            a = "ff"
+        elif hex_len == 9:
+            r = color_hex[1:3]
+            g = color_hex[3:5]
+            b = color_hex[5:7]
+            a = color_hex[7:9]
+        else:
+            raise InvalidColorException(color)
+
+        try:
+            color = int(r, 16), int(g, 16), int(b, 16), int(a, 16)
+        except:
+            raise InvalidColorException(color)
+
+    if is_color_tuple_like(color):
+        color_tuple = cast(ColorTuple, color)
+        return _normalize_tuple(color_tuple, rgb_formatter, alpha_formatter)
+
+    raise InvalidColorException(color)
+
+
+def _normalize_tuple(
+    color: ColorTuple,
+    rgb_formatter: Callable[[float, MaybeColor], float],
+    alpha_formatter: Callable[[float, MaybeColor], float],
+) -> ColorTuple:
+    """Parse color tuple using the specified color formatters.
+
+    The R, G, B components are transformed by rgb_formatter, and the alpha component is transformed
+    by alpha_formatter.
+
+    For example, to output a (float, float, float, int) color tuple, set rgb_formatter
+    to _float_formatter and alpha_formatter to _int_formatter.
+    """
+    if len(color) == 3:
+        r = rgb_formatter(color[0], color)
+        g = rgb_formatter(color[1], color)
+        b = rgb_formatter(color[2], color)
+        return r, g, b
+
+    elif len(color) == 4:
+        color_4tuple = cast(Color4Tuple, color)
+        r = rgb_formatter(color_4tuple[0], color_4tuple)
+        g = rgb_formatter(color_4tuple[1], color_4tuple)
+        b = rgb_formatter(color_4tuple[2], color_4tuple)
+        alpha = alpha_formatter(color_4tuple[3], color_4tuple)
+        return r, g, b, alpha
+
+    raise InvalidColorException(color)
+
+
+def _int_formatter(component: float, color: MaybeColor) -> int:
+    """Convert a color component (float or int) to an int from 0 to 255.
+
+    Anything too small will become a 0, and anythign too large will become 255.
+    """
+    if isinstance(component, float):
+        component = int(component * 255)
+
+    if isinstance(component, int):
+        return min(255, max(component, 0))
+
+    raise InvalidColorException(color)
+
+
+def _float_formatter(component: float, color: MaybeColor) -> float:
+    """Convert a color component (float or int) to a float from 0.0 to 1.0.
+
+    Anything too small will become a 0.0, and anythign too large will become 1.0.
```

Comment:
nit:
`Anything too small will become a 0.0, and anything too large will become 1.0.`

Reply:
Oops, only saw this after merging. Will fix in my next few PRs.